interface/ceed-elemrestriction.c

```diff
@@ -440,8 +538,80 @@ int CeedElemRestrictionCreateOriented(Ceed ceed, CeedInt num_elem, CeedInt elem_
   (*rstr)->l_size      = l_size;
   (*rstr)->num_blk     = num_elem;
   (*rstr)->blk_size    = 1;
-  (*rstr)->is_oriented = 1;
-  CeedCall(ceed->ElemRestrictionCreateOriented(mem_type, copy_mode, offsets, orient, *rstr));
+  CeedCall(ceed->ElemRestrictionCreateOriented(mem_type, copy_mode, offsets, orients, *rstr));
+  return CEED_ERROR_SUCCESS;
+}
+
+/**
+  @brief Create a CeedElemRestriction with a general tridiagonal transformation matrix for curl-conforming elements
+
+  @param[in]  ceed         Ceed object where the CeedElemRestriction will be created
+  @param[in]  num_elem     Number of elements described in the @a offsets array
+  @param[in]  elem_size    Size (number of "nodes") per element
+  @param[in]  num_comp     Number of field components per interpolation node (1 for scalar fields)
+  @param[in]  comp_stride  Stride between components for the same L-vector "node". Data for node i, component j, element k can be found in the
+L-vector at index offsets[i + k*elem_size] + j*comp_stride.
+  @param[in]  l_size       The size of the L-vector. This vector may be larger than the elements and fields given by this restriction.
+  @param[in]  mem_type     Memory type of the @a offsets array, see CeedMemType
+  @param[in]  copy_mode    Copy mode for the @a offsets array, see CeedCopyMode
+  @param[in]  offsets      Array of shape [@a num_elem, @a elem_size]. Row i holds the ordered list of the offsets (into the input CeedVector) for the
+unknowns corresponding to element i, where 0 <= i < @a num_elem. All offsets must be in the range [0, @a l_size - 1].
+  @param[in]  curl_orients Array of shape [@a num_elem, @a 3 * elem_size] representing a row-major tridiagonal matrix (curl_orients[0] =
+curl_orients[(i + 1) * 3 * elem_size - 1] = 0, where 0 <= i < @a num_elem) which is applied to the element unknowns upon restriction.
```

Comment:
I would be open to any suggestions you have for improving the docstring. I agree it is hard to explain without giving a lot of background information, but I think users can resolve any further misunderstanding by just looking at the unit tests for example usage.

Reply:
Why not go ahead and provide some background information?  It's my understanding that there's not necessarily one "standard" way to handle this among libraries that can do H(curl) elements (or perhaps because it would normally be handled completely inside the library, and not require a parameter from the user), so that's why I think we need to be very clear about what the input needs to be. I don't really like the idea of asking users to parse out the proper way to use a parameter by reading the tests.  Though I suppose adding some more detailed comments to the tests would be better than nothing, and make them even more useful for helping users understand. Even just mentioning a bit of context about the tridiagonal format coming from the pairs of face DOFs would be helpful, IMO.

Does anyone else have any thoughts about whether we should have some more detailed information about element restrictions (basic, oriented, curl oriented) in the higher-level docs (like in doc/sphinx/source/libCEEDapi.md or elsewhere)?  Or is it unnecessary?  If we're going to add that in the future it wouldn't need to be in the PR necessarily.

Reply:
Hi @nbeams, check out 268839c. Is this satisfactory? I referenced the above paper as well.

Reply:
Thanks, I think that's really helpful to have 👍